src/reader-activation/analytics.js

```diff
@@ -1,16 +1,16 @@
 /**
- * Get a GA4 event payload for a given prompt.
+ * Get a GA4 event payload.
  *
- * @param {string} action      Action name for the event.
- * @param {number} promptId    ID of the prompt
- * @param {Object} extraParams Additional key/value pairs to add as params to the event payload.
+ * @param {Object} payload Event payload.
+ * @param {Object} data    Data from the dispatched reader data activity.
  *
  * @return {Object} Event payload.
  */
-
-const getEventPayload = ( extraParams = {} ) => {
+const getEventPayload = ( payload = {}, data = {} ) => {
 	return {
-		...extraParams,
+		...payload,
+		newspack_popup_id: data?.newspack_popup_id || null,
+		gate_post_id: data?.gate_post_id || null,
```

Comment:
We might want to avoid adding these keys to the payload if the values are `null` to a.) save a few bytes per GA request, and b.) because of GA's limit of 25 max params per event, these empty params are just taking up space.

Reply:
Good call! Fixed in ac05a08e338717472adbe548a7d545487470ead7 and also removed the `referrer` parameter (see p1746557784095789-slack-C015W6BES8J)